src/searchengine/nova/engines/torrentz.py

```diff
@@ -0,0 +1,113 @@
+#VERSION: 2.11
+#AUTHORS: Diego de las Heras (diegodelasheras@gmail.com)
+
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+#    * Redistributions of source code must retain the above copyright notice,
+#      this list of conditions and the following disclaimer.
+#    * Redistributions in binary form must reproduce the above copyright
+#      notice, this list of conditions and the following disclaimer in the
+#      documentation and/or other materials provided with the distribution.
+#    * Neither the name of the author nor the names of its contributors may be
+#      used to endorse or promote products derived from this software without
+#      specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+from novaprinter import prettyPrinter
+from helpers import retrieve_url, download_file
+from HTMLParser import HTMLParser
+from urllib import urlencode
+
+class torrentz(object):
+    # mandatory properties
+    url  = 'https://torrentz.eu'
+    name = 'Torrentz'
+    supported_categories = {'all': ''}
+
+    trackers_list = ['udp://open.demonii.com:1337/announce',
+                    'udp://tracker.openbittorrent.com:80/announce',
+                    'udp://tracker.publicbt.com:80/announce',
+                    'udp://tracker.istole.it:80/announce']
```

Comment:
Torrentz doesn't provide torrents or magnets, just give you links to other search engines. I'm generating the magnet link with the hash and name from torrentz and a public trackers list.

Example:
https://torrentz.eu/cb84ccc10f296df72d6c40ba7a07c178a4323a14